bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/form/FormPostServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+        try {
+            JSONObject jsonRequest = requestParamsToJSON(request);
+            return processFormApi(request, response, jsonRequest);
+        } catch (JSONException e) {
+            LOG.error(e.getMessage());
+        }
+        return false;
+    }
+
+    private boolean processFormApi(SlingHttpServletRequest request, SlingHttpServletResponse response, JSONObject jsonRequest) {
+        boolean processFormApiSuccess = false;
+        Resource formContainerResource = request.getResource();
+        ValueMap valueMap = formContainerResource.adaptTo(ValueMap.class);
+        if (valueMap != null) {
+            String formEndpoint = valueMap.get(PN_FORM_ENDPOINT_URL, String.class);
+            if (StringUtils.isNotEmpty(formEndpoint)) {
+                processFormApiSuccess = callFormApi(formEndpoint, jsonRequest);
+            }
+            sendRedirect(valueMap, request, response, processFormApiSuccess);
+        }
+        return processFormApiSuccess;
+    }
+
+    private void sendRedirect(ValueMap valueMap, SlingHttpServletRequest request, SlingHttpServletResponse response, boolean processFormApiSuccess) {
+        String redirect = valueMap.get("redirect", String.class);
+        try {
+            if (!StringUtils.isEmpty(redirect) && processFormApiSuccess) {
+                if (!redirect.contains(".")) {
+                    redirect = redirect + ".html";
+                }
+                response.sendRedirect(request.getResourceResolver().map(request, redirect));
+            } else {
+                FormsHelper.redirectToReferrer(request, response, new HashMap<>());
+            }
+        } catch (IOException var13) {
+            LOG.error("Error redirecting to {}", redirect);
+        }
+    }
+
+    private boolean callFormApi(String formEndpointUrl, JSONObject jsonRequest) {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(connectionTimeout)
+                .setConnectionRequestTimeout(connectionTimeout)
+                .setSocketTimeout(socketTimeout)
+                .build();
+
+        CloseableHttpClient client = clientBuilderFactory.newBuilder().setDefaultRequestConfig(config).build();
+        HttpPost post = new HttpPost(formEndpointUrl);
+        post.setEntity(new StringEntity(jsonRequest.toString(), ContentType.create(JSONResponse.RESPONSE_CONTENT_TYPE,
+                CHARSET)));
+        CloseableHttpResponse response = null;
+        try {
+            response = client.execute(post);
+        } catch (IOException e) {
+            LOG.error(e.getMessage());
+        }
+        StatusLine responseStatusLine = response != null ? response.getStatusLine() : null;
+        return responseStatusLine != null && responseStatusLine.getStatusCode() == HttpStatus.SC_OK;
+    }
+
+    private JSONObject requestParamsToJSON(SlingHttpServletRequest req) throws JSONException {
```

Comment:
I'd rename it `getFormData`. It could be part of the FormUtils mentioned in the PR.